indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts

```diff
@@ -213,23 +229,305 @@ export class AggregateTradingReward {
+        return {
+          id: TradingRewardAggregationTable.uuid(address, this.period, startedAtHeight),
+          endedAt,
+          endedAtHeight,
+        };
+      },
+    );
+    return {
+      createObjects: _.map(
+        aggregationUpdateAndCreateObjects.createObjects,
+        (createObject: TradingRewardAggregationCreateObject) => {
+          return {
+            ...createObject,
+            endedAt,
+            endedAtHeight,
+          };
+        },
+      ),
+      updateObjects: _.map(
+        aggregationUpdateAndCreateObjects.updateObjects,
+        (updateObject: TradingRewardAggregationUpdateObject) => {
+          return {
+            ...updateObject,
+            endedAt,
+            endedAtHeight,
+          } as TradingRewardAggregationUpdateObject;
+        },
+      ).concat(otherAggregationUpdateObjects),
+    };
+  }
+
+  private getEndedAt(interval: Interval): IsoString {
+    const startedAt: IsoString = this.getStartedAt(interval);
+
+    return DateTime
+      .fromISO(startedAt, UTC_OPTIONS)
+      .endOf(this.getDateTimeUnit())
+      .plus({ milliseconds: 1 })
+      .toISO();
+  }
+
+  private getDateTimeUnit(): DateTimeUnit {
+    switch (this.period) {
+      case TradingRewardAggregationPeriod.DAILY:
+        return DateTimeUnit.day;
+      case TradingRewardAggregationPeriod.WEEKLY:
+        return DateTimeUnit.week;
+      case TradingRewardAggregationPeriod.MONTHLY:
+        return DateTimeUnit.week;
+      default:
+        throw new Error(`Invalid period ${this.period}`);
+    }
+  }
+
+  private async setAggregationUpdateAndCreateObjects(
+    aggregationUpdateAndCreateObjects: AggregationUpdateAndCreateObjects,
+  ): Promise<void> {
+    const createObjectsChunks: TradingRewardAggregationCreateObject[][] = _.chunk(
+      aggregationUpdateAndCreateObjects.createObjects,
+      config.AGGREGATE_TRADING_REWARDS_CHUNK_SIZE,
+    );
+    for (const createObjectsChunk of createObjectsChunks) {
+      logger.info({
+        at: 'aggregate-trading-rewards#setAggregationUpdateAndCreateObjects',
+        message: 'Creating trading reward aggregations',
+        count: createObjectsChunk.length,
+        createObjectsChunk: JSON.stringify(createObjectsChunk),
+      });
+      await Promise.all(_.map(
+        createObjectsChunk,
+        (createObject: TradingRewardAggregationCreateObject) => {
+          return TradingRewardAggregationTable.create(createObject);
+        },
+      ));
+      logger.info({
+        at: 'aggregate-trading-rewards#setAggregationUpdateAndCreateObjects',
+        message: 'Created trading reward aggregations',
+        count: createObjectsChunk.length,
+      });
+    }
+
+    const updateObjectsChunks: TradingRewardAggregationUpdateObject[][] = _.chunk(
+      aggregationUpdateAndCreateObjects.updateObjects,
+      config.AGGREGATE_TRADING_REWARDS_CHUNK_SIZE,
+    );
+    for (const updateObjectsChunk of updateObjectsChunks) {
+      logger.info({
```

Comment:
should we make this a stat instead?

Reply:
added a stat to track each chunk, I kept the logs since theyre more for debugging purposes